src/AggregateRoots/AggregateEntity.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Spatie\EventSourcing\AggregateRoots;
+
+use Spatie\EventSourcing\EventHandlers\AppliesEvents;
+use Spatie\EventSourcing\StoredEvents\ShouldBeStored;
+use Spatie\EventSourcing\StoredEvents\StoredEvent;
+
+abstract class AggregateEntity
+{
+    use AppliesEvents;
+
+    public function __construct(
+        private AggregateRoot $aggregateRoot
```

Comment:
I'm thinking we could use `protected` here

Reply:
Great idea!

Reply:
Just tried it, unfortunately we can't do it because an AR is resolved via the container, so the constructor must be public.